src/meta/src/hummock/compaction/tier_compaction_picker.rs

```diff
@@ -196,3 +196,63 @@ impl CompactionPicker for TierCompactionPicker {
         self.pick_whole_level(l0, &level_handlers[0])
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use std::sync::Arc;
+
+    use risingwave_pb::hummock::hummock_version::Levels;
+    use risingwave_pb::hummock::LevelType;
+
+    use crate::hummock::compaction::compaction_config::CompactionConfigBuilder;
+    use crate::hummock::compaction::level_selector::tests::{
+        generate_l0_overlapping_sublevels, generate_level, generate_table,
+    };
+    use crate::hummock::compaction::overlap_strategy::RangeOverlapStrategy;
+    use crate::hummock::compaction::{CompactionPicker, TierCompactionPicker};
+    use crate::hummock::level_handler::LevelHandler;
+
+    #[test]
+    fn test_pick_whole_level_skip_sublevel() {
+        let l0 = generate_l0_overlapping_sublevels(vec![
+            vec![
+                generate_table(4, 1, 10, 90, 1),
+                generate_table(5, 1, 210, 220, 1),
+            ],
+            vec![generate_table(6, 1, 0, 100, 1)],
+            vec![generate_table(7, 1, 0, 100, 1)],
```

Comment:
These are non-overlapping sub-levels. Is using `generate_l0_overlapping_sublevels` expected?

Reply:
Yes, here we construct a L0 with only the top level being non-overlapping, by explicitly setting sub-level 0 to non-overlapping later.